content/blog/introducing-crd2pulumi/index.md

```diff
@@ -0,0 +1,320 @@
+complex YAML schemas, making it difficult to write CustomResources that adhere to those specs. Programming languages
+offer a better path forward. Instead of wrangling error-prone YAML definitions, using types in a programming language
+lets you use IDE type checking and autocomplete features!
+
+## Getting Started with crd2pulumi
+
+Let's test `crd2pulumi` on the example [CronTab CRD] specified in the Kubernetes Documentation.
+
+```yaml
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  # name must match the spec fields below, and be in the form: <plural>.<group>
+  name: crontabs.stable.example.com
+spec:
+  # group name to use for REST API: /apis/<group>/<version>
+  group: stable.example.com
+  # list of versions supported by this CustomResourceDefinition
+  versions:
+    - name: v1
+      # Each version can be enabled/disabled by Served flag.
+      served: true
+      # One and only one version must be marked as the storage version.
+      storage: true
+      schema:
+        openAPIV3Schema:
+          type: object
+          properties:
+            spec:
+              type: object
+              properties:
+                cronSpec:
+                  type: string
+                image:
+                  type: string
+                replicas:
+                  type: integer
+  # either Namespaced or Cluster
+  scope: Namespaced
+  names:
+    # plural name to be used in the URL: /apis/<group>/<version>/<plural>
+    plural: crontabs
+    # singular name to be used as an alias on the CLI and for display
+    singular: crontab
+    # kind is normally the CamelCased singular type. Your resource manifests use this.
+    kind: CronTab
+    # shortNames allow shorter string to match your resource on the CLI
+    shortNames:
+    - ct
+```
+
+Create a file called `crd.yaml` with this definition and then run `crd2pulumi` for your language of choice.
+
+{{% notes type="info" %}}
+crd2pulumi supports TypeScript and Go today, and Python and .NET is coming soon.
+{{% /notes %}}
+
+By default, this creates the folder `crontabs/` in the same directory as the YAML file. Now you can instantiate a
+Pulumi project and import the generated code. In this example, we register the CronTab CRD and then create a CronTab
+CustomResource.
+
+{{< chooser language "typescript,python,csharp,go" >}}
+
+{{% choosable language typescript %}}
+
+```
+$ crd2pulumi nodejs crd.yaml
+```
+
+```typescript
+import * as crontabs from "./crontabs"
+import * as pulumi from "@pulumi/pulumi"
+
+// Register the CronTab CRD.
+const cronTabDefinition = new crontabs.CronTabDefinition("my-crontab-definition");
+
+// Instantiate a CronTab resource.
+const myCronTab = new crontabs.v1.CronTab("my-new-cron-object",
+{
+	metadata: {
+    	name: "my-new-cron-object",
+	},
+	spec: {
+    	cronSpec: "* * * * */5",
+    	image: "my-awesome-cron-image",
+        replicas: 3,
+	}
+});
```

Comment:
Nit: There was a mix of tabs/spaces for indentation. This fixes it to use spaces consistently (which is what we typically use for indentation for TS elsewhere):

```suggestion
const myCronTab = new crontabs.v1.CronTab("my-new-cron-object",
{
    metadata: {
        name: "my-new-cron-object",
    },
    spec: {
        cronSpec: "* * * * */5",
        image: "my-awesome-cron-image",
        replicas: 3,
    }
});
```